File: valid-parentheses/valid-parentheses.cpp
class Solution {

bool match(char close,char open){
 if( ( open =='(' && close == ')') || ( open =='[' && close == ']') || (open =='{' && close == '}') )
    return true;
    

    return false;
}

public:
    bool isValid(string s) {

        if(s.length()&1){
            return false;
        }

        stack<char>st;
        
        for(int i=0;i<s.length();i++){
            char ch= s[i];
            if(ch=='(' || ch == '{' || ch == '[')
                st.push(ch);
            else{
                if(!st.empty() && match(ch,st.top()))
                st.pop();
                else
                return false;
            }
        }

        if(st.empty())
        return true;

        return false;
    }
};

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {

     

         Stack<Character> st = new Stack<Character>(); 

         for (char c : s.toCharArray()) {
            if(c == '(' || c == '[' || c == '{'){
            st.push(c);

            }else
            {
                 if(st.isEmpty()){
            return false;
            }

              char top = st.peek();
              if ((c == ')' && top == '(') || (c == ']' && top == '[') || (c == '}' && top == '{')) {
                    // If it is, pop the opening bracket from the stack
                    st.pop();
            }else
            return false;
         }}


          return st.isEmpty();
    }
}